public/app/features/auth-config/fields.ts

```diff
@@ -19,13 +19,81 @@ export const fields: Record<SSOProvider['provider'], Array<keyof SSOProvider['se
   ],
 };
 
+type Section = Record<
+  SSOProvider['provider'],
+  Array<{
+    name: string;
+    id: string;
+    fields: SSOSettingsField[];
+  }>
+>;
+
+export const sectionFields: Section = {
+  generic_oauth: [
+    {
+      name: 'General settings',
+      id: 'general',
+      fields: [
+        'name',
+        'clientId',
+        'clientSecret',
+        'scopes',
+        'authUrl',
+        'tokenUrl',
+        'apiUrl',
+        'allowSignUp',
+        'autoLogin',
+        'signoutRedirectUrl',
+      ],
+    },
+    {
+      name: 'User mapping',
+      id: 'user',
+      fields: [
+        'nameAttributePath',
+        'loginAttributePath',
+        'emailAttributeName',
+        'emailAttributePath',
+        'idTokenAttributeName',
+        'roleAttributePath',
+        'roleAttributeStrict',
+        'allowAssignGrafanaAdmin',
+        'skipOrgRoleSync',
+      ],
+    },
+    {
+      name: 'Extra security measures',
+      id: 'extra',
+      fields: [
+        'allowedOrganizations',
+        'allowedDomains',
+        'defineAllowedGroups',
+        { name: 'allowedGroups', dependsOn: 'defineAllowedGroups' },
+        { name: 'groupsAttributePath', dependsOn: 'defineAllowedGroups' },
+        'defineAllowedTeamsIds',
+        { name: 'teamIds', dependsOn: 'defineAllowedTeamsIds' },
+        { name: 'teamsUrl', dependsOn: 'defineAllowedTeamsIds' },
+        { name: 'teamIdsAttributePath', dependsOn: 'defineAllowedTeamsIds' },
+        'usePkce',
+        'useRefreshToken',
+      ],
+    },
+    {
+      name: 'TLS',
+      id: 'tls',
+      fields: ['configureTLS', 'tlsSkipVerifyInsecure', 'tlsClientCert', 'tlsClientKey', 'tlsClientCa'],
+    },
+  ],
+};
+
 /**
  * List all the fields that can be used in the form
  */
 export const fieldMap: Record<string, FieldData> = {
```

Comment:
@lwandz13 Could you please review the `description` fields for the generic_oauth settings in this file? For more info you can find more info about the different fields on https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/generic-oauth/#configuration-options

Thanks